### ISSUE TITLE
v0.5 - custom authorizer issue

### ISSUE THREAD

**Issue by ajagnanan** [serverless/serverless#797]
My current setup: 

Add the following to the custom authorizer function: 

``` JSON
"name": "custom_authorizer",
...
"authorizer": {
     "identitySource": "method.request.header.Authorization"
 },
...
```

Add this to endpoint: 

``` JSON
...
 "authorizationType": "custom",
 "authorizerFunction": "custom_authorizer",
...
```

Result: 
The custom authorizer is created and the endpoint references it, it is missing the region and lambda function name though. It seems that in `EndpointDeployApiGateway.js` the alias isn't resolved at this line: 

```
// Construct authorizer URI
let alias = '${stageVariables.functionAlias}';
```

Workaround:
Comment out the following lines for now:

```
//+ ':'
//+ alias
```

in `_updateAuthorizer` and `_createAuthorizer`

**Comment by neaghfoz** [comment]
Same problem here but comment that lines it throw exception: ```BadRequestException: Invalid lambda function```

**Comment by ac360** [comment]
Thanks @ajagnanan. Is the problem only in the AWS dashboard or does this not work when you hit the endpoint outside of the dashboard?  Please know that the AWS dashboard does not currently work w/ authorizers that use Lambdas w/ stageVariables.  They need to update their GUI badly.  Things will appear not to work :(

**Comment by ajagnanan** [comment]
Yea, this is an external call via an angular app. It doesn't really show an error message, but throws a 500 status code and these headers are returned: 

```
x-amzn-ErrorType:AuthorizerConfigurationException
x-amzn-RequestId:c11e9305-eab3-11e5-9be0-d79fa3ee548e
```

**Comment by ac360** [comment]
Thanks @ajagnanan.  Looking into this now

**Comment by michaelorionmcmanus** [comment]
@ac360 @ajagnanan just took the leap to manage a custom authorizer as a serverless function and I thought I should share my findings on this thread. 

The point made by @ajagnanan is a bit of a red herring but understandably confusing. The value of `alias` is correct and will be expanded by api gateway during invocation. API Gateway's stage variable  syntax is pretty much identical to serverless template variables. 

The issue I ran into is that the authorizer create and update operations deal with the authorizer's `authorizerCredentials` property somewhat ambiguously. To get the current implementation working I had to supply the role that the authorizer should execute with, during creation. 

I have a PR that adds support for updating this property via a patch operation which is minimally what I had to alter to get the custom authorizer working, in my case: https://github.com/serverless/serverless/pull/818

And yes the apig GUI is a source of pain :cry: It lies. To see that your changes are taking effect when you deploy your endpoint you're better off using the cli and run something like the following command in order to inspect your authorizers:

```
aws apigateway get-authorizers --rest-api-id <API_ID> --region <REGION> --profile <PROFILE>
```
